internal/analysis/analysis.go

```diff
@@ -98,6 +99,28 @@ var (
 	resultError = &Result{Status: StatusErrorOther}
 )
 
+// RunAllPhases
+// Runs the given phases of dynamic analysis (e.g. import, install) of the given package
+// in the sandbox provided, and returns the result.
+func RunAllPhases(sb sandbox.Sandbox, pkg *pkgecosystem.Pkg) (map[pkgecosystem.RunPhase]*Result, error) {
+	results := make(map[pkgecosystem.RunPhase]*Result)
+	phases := pkg.Manager().RunPhases()
+	for _, phase := range phases {
+		result, err := Run(sb, pkg.Command(phase))
+		results[phase] = result
+		if err != nil || result.Status != StatusCompleted {
```

Comment:
@calebbrown does this error handling logic seem alright?

Reply:
I left a comment in chat.

Errors are intended for problems with trying to do the dynamic analysis. i.e. podman fails, the disk is full, etc. In this scenario, no result is captured as the data will be incomplete or missing.

result.Status != StatusCompleted is intended to capture that the while we did some work there was some other failure - e.g. installation failed, etc. In this scenario data is complete and a result can be captured.

In time we may want to clean this up, but right now the current implementation changes the behavior by capturing a result when `err != nil`.

Additionally the log messages feed into our metrics dashboard, so ideally they should be preserved.

Reply:
Thanks, it helps a lot to have this higher level understanding. I'll add comments to capture it in the code.